net/http/authn/authn.go

```diff
@@ -72,8 +80,14 @@ func (a *API) Authenticate(req *http.Request) (*http.Request, error) {
 
 	// if there is no authentication at all, we return an "unauthenticated" error,
 	// which may be helpful when debugging
-	if len(X509Certs(ctx)) < 1 && err != nil {
-		return req, errors.New("unauthenticated")
+	if len(X509Certs(ctx)) < 1 && Token(ctx) == "" {
+		err := errors.New("unauthenticated")
+		if len(authnErrors) > 0 {
+			err = errors.WithDetailf(err, "Invalid credentials: %s", strings.Join(authnErrors, "; "))
```

Comment:
Ah, I see. The "Detail: " prefix is added by corectl. Maybe we should get rid of those prefixes from corectl? 

```
RPC error: CH009 Request could not be authenticated
Detail: Invalid credentials: invalid token: "jackattack"
```
vs
```
CH009 Request could not be authenticated
Invalid credentials: invalid token: "jackattack"
```